games-rpg/dreamweb/dreamweb-1.1-r1.ebuild

```diff
@@ -16,50 +17,30 @@ SRC_URI="doc? ( mirror://sourceforge/scummvm/${PN}-manuals-en-highres.zip )
 	!l10n_de? ( !l10n_en? ( !l10n_en-GB? ( !l10n_es? ( !l10n_fr? ( !l10n_it? \
 		( mirror://sourceforge/scummvm/${PN}-cd-us-${PV}.zip ) ) ) ) ) )
 	http://www.scummvm.org/images/cat-dreamweb.png"
+S="${WORKDIR}"
 
 LICENSE="Dreamweb"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE="doc l10n_de l10n_en l10n_en-GB l10n_es l10n_fr l10n_it"
+IUSE="doc l10n_de +l10n_en l10n_en-GB l10n_es l10n_fr l10n_it"
 
-RDEPEND=">=games-engines/scummvm-1.7[flac]"
-DEPEND="app-arch/unzip"
-
-S="${WORKDIR}"
+RDEPEND="games-engines/scummvm[flac]"
+BDEPEND="app-arch/unzip"
 
 src_unpack() {
-	if use l10n_de ; then
-		mkdir -p "${S}"/de || die
-		cd "${S}"/de || die
-		unpack ${PN}-cd-de-${PV}.zip
-	fi
-	if use l10n_en || ( ! use l10n_de && ! use l10n_en && ! use l10n_en-GB && \
-			! use l10n_es && ! use l10n_fr && ! use l10n_it ) ; then
-		mkdir -p "${S}"/en_US || die
-		cd "${S}"/en_US || die
-		unpack ${PN}-cd-us-${PV}.zip
-	fi
-	if use l10n_en-GB ; then
-		mkdir -p "${S}"/en_GB || die
-		cd "${S}"/en_GB || die
-		unpack ${PN}-cd-uk-${PV}.zip
-	fi
-	if use l10n_es ; then
-		mkdir -p "${S}"/es || die
-		cd "${S}"/es || die
-		unpack ${PN}-cd-es-${PV}.zip
-	fi
-	if use l10n_fr ; then
-		mkdir -p "${S}"/fr || die
-		cd "${S}"/fr || die
-		unpack ${PN}-cd-fr-${PV}.zip
-	fi
-	if use l10n_it ; then
-		mkdir -p "${S}"/it || die
-		cd "${S}"/it || die
-		unpack ${PN}-cd-it-${PV}.zip
-	fi
-	if use doc ; then
+	MY_L10N=( $(usev l10n_de) $(usev l10n_es) $(usev l10n_fr) $(usev l10n_it) )
+	use l10n_en && MY_L10N+=( l10n_us )
+	use l10n_en-GB && MY_L10N+=( l10n_uk )
+	[[ ${MY_L10N} ]] || MY_L10N=( l10n_us )
+
+	local lang
+	for lang in "${MY_L10N[@]//l10n_/}"; do
+		mkdir -p "${S}"/${lang} || die
```

Comment:
Drop the `-p` everywhere (including other ebuilds), it would be strange if "${S}" didn't already exist and this was somehow needed.